server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -2234,6 +2240,22 @@ public SeqNoStats getSeqNoStats(long globalCheckpoint) {
         return localCheckpointTracker.getStats(globalCheckpoint);
     }
 
+    @Override
+    public Engine lockDownEngine() throws IOException {
```

Comment:
ideally we would do this entirely outside of the engine and maybe just pass and engine to the ctor of ReadOnlyEngine? do we need to make sure we don't receive writes after we did this or why do we acquire a write lock?

Reply:
I will move this to ctor of ReadOnlyEngine.